parsons/google/google_cloud_storage.py

```diff
@@ -568,4 +606,26 @@ def unzip_blob(
                     local_path=decompressed_filepath,
                 )
 
-        return self.format_uri(bucket=bucket_name, name=decompressed_blob_name)
+    def __zip_decompress_and_write_to_gcs(self, **kwargs):
+        compressed_filepath = kwargs.pop("compressed_filepath")
+        decompressed_blob_name = kwargs.pop("decompressed_blob_name")
+        decompressed_blob_in_archive = decompressed_blob_name.split("/")[-1]
+        bucket_name = kwargs.pop("bucket_name")
+        new_file_extension = kwargs.pop("new_file_extension")
+
+        staging_file = "./staging"
```

Comment:
I'd like to do something different here

Reply:
As I understand the `zipfile` process here we need to (1) unzip the archive (2) open the file to be read, which has the same filename as the decompressed blob and (3) copy the contents into a file to be put in storage

If we could combine steps 2 and 3 and simply stream the contents of the file to GCS that would obviously be preferable, but this current iteration is functional at the very least

Reply:
love this suggestion!